engine/src/main/java/org/terasology/config/flexible/SettingImpl.java

```diff
@@ -53,8 +60,8 @@
      * @param id           the id of the setting.
      * @param defaultValue the default value of the setting.
      */
-    public SettingImpl(SimpleUri id, T defaultValue) {
-        this(id, defaultValue, null);
+    public SettingImpl(Class<T> valueClass, SimpleUri id, T defaultValue) {
+        this(valueClass, id, defaultValue, new DefaultValueValidator<>());
```

Comment:
The SettingValueValidator in a Setting is publicly accessible (through `getValidator`), and may be used in a number of places (for example, in UI code to check whether a value can be assigned to the setting or not). As such, I would prefer consumers to not have to worry about the possibility of the validator being null, thus avoiding a potential NPE.

Reply:
Addendum: after being reminded that `Optional` does exist in Java, a better option might be to use an `Optional<SettingValueValidator<T>>` instead.

Reply:
I have discussed this issue with a couple of people (Skaldarnar, vampcat). The conclusion we came to is that in this particular case, so far, using Optional isn't worth the effort. If we had a number of chainable potentially null objects it would make sense. In this case nothing else can be null, only the validator. 

Also, a DefaultValueValidator ofuscates the fact that it isn't, in fact, a validator: it doesn't do anything. In this context I prefer to keep null as a possible return value of getValidator, with the implied meaning that there is no validator. As long as this is clearly documented in the method's javadoc it's enough.